Heads up — `code-quality` is failing on a real mypy error (blocks the required gate), not a flake:

```
autobot_shared/plugin_sdk/loader.py:297: error: Incompatible types in assignment (expression has type "object", variable has type Module | None)  [assignment]
```

`module = self._import_from_file(entry_point, plugin_dir)` — `_import_from_file` is inferred to return `object`, but `module` is `ModuleType | None`. Annotate `_import_from_file`'s return as `ModuleType | None` (or cast at the call site) and code-quality should go green. Left unmerged for you since your worktree is active on `issue-10294`.

--- a/autobot_shared/plugin_sdk/loader.py
+++ b/autobot_shared/plugin_sdk/loader.py
@@ -19,6 +19,7 @@ import json
 import logging
 import os
 from pathlib import Path
+from types import ModuleType
 from typing import Dict, List, Tuple, Type
 
 from .base import BasePlugin, PluginLoadError, PluginManifest, PluginRegistry, PluginStatus
@@ -309,7 +310,7 @@ class PluginLoader:
         logger.error("No plugin class found in module: %s", entry_point)
         return None
 
-    def _import_from_file(self, entry_point: str, plugin_dir: Path | None) -> object | None:
+    def _import_from_file(self, entry_point: str, plugin_dir: Path | None) -> ModuleType | None:
         """
         File-path fallback for plugins whose directory name is not importable
         (e.g. ``core-plugins/image-generation-plugin`` -- hyphen prevents normal import).

